### PR TITLE
Add separate default options for emojione 3

### PR DESCRIPTION
**Pull request by pederjohnsen**
So I can easier work on this outside of work I'm replacing https://github.com/mervick/emojionearea/pull/175 with this PR from my personal github account.

**[Last comment from previous PR]**
@mervick I made some changes, so detectVersion and getSupportMode can be used outside of loadEmojione, this way we can use the emojione var to get the correct support mode in getDefaultOptions when it is run the very first time ensuring we get the right default options. What do you think?

@lasdeparq do you know if the list of emojis is 100% accurate/complete? (I just used the ones from your fiddle)

### CODE REVIEW

**Comment by carpii** [comment]
Looks like you left a few debug logs in...

````
console.log('#####hmmm2');
console.log('getDefaultOptions');
````

**Comment by lasdeparq** [comment]
@johnsenpeder , I've reached the list by getting by categories on API of the emojione lib.
Maybe one or another is missing, but virtually everyone is there.

File: dist/emojionearea.js [hunk 1]
```diff
@@ -131,8 +131,51 @@ document = window.document || {};
...
+            case '3.1.0':
+            case '3.1.1':
+            case '3.1.2':
+            case '3.1':
+            default: return 6;
+        }
+    };
     var getDefaultOptions = function () {
-        return $.fn.emojioneArea && $.fn.emojioneArea.defaults ? $.fn.emojioneArea.defaults : {
+        console.log('#####hmmm2');
```
Comment (mervick) [inline dist/emojionearea.js]: remove this

File: dist/emojionearea.js [hunk 2]
```diff
@@ -131,8 +131,51 @@ document = window.document || {};
...
+        console.log('getDefaultOptions');
```
Comment (mervick) [inline dist/emojionearea.js]: remove this

File: dist/emojionearea.js [hunk 3]
```diff
@@ -164,9 +207,240 @@ document = window.document || {};
             textcomplete: {
                 maxCount      : 15,
                 placement     : null // null - default | top | absleft | absright
-            },
+            }
+        };
+
+        console.log('#####TEST');
```
Comment (mervick) [inline dist/emojionearea.js]: remove this

File: dist/emojionearea.js [hunk 4]
```diff
@@ -164,9 +207,240 @@ document = window.document || {};
...
+        console.log(getSupportMode(detectVersion(emojione)));
```
Comment (mervick) [inline dist/emojionearea.js]: remove this

File: src/var/getDefaultOptions.js [hunk 5]
```diff
@@ -1,9 +1,12 @@
...
+    'jquery',
+    'var/emojione',
+    'function/detectVersion',
+    'function/getSupportMode'
 ],
-function($) {
+function($, emojione, detectVersion, getSupportMode) {
     return function () {
-        return $.fn.emojioneArea && $.fn.emojioneArea.defaults ? $.fn.emojioneArea.defaults : {
+        var defaultOptions = {
```
Comment (mervick) [inline src/var/getDefaultOptions.js]: There is no need to init `defaultOptions` when `$.fn.emojioneArea.defaults` exists.
So need to check `$.fn.emojioneArea.defaults ` first and after that init `defaultOptions`
```js
if ($.fn.emojioneArea && $.fn.emojioneArea.defaults) {
    return fn.emojioneArea.defaults;
}
var defaultOptions = {
// ...
```

File: src/var/getDefaultOptions.js [hunk 6]
```diff
@@ -208,7 +439,9 @@ function($) {
                     "tg to tt tn tr flag_tm flag_tm ug ua ae gb us vi uy uz vu va ve vn wf eh ye zm zw re ax ta io bq cx " +
                     "cc gg im yt nf pn bl pm gs tk bv hm sj um ic ea cp dg as aq vg ck cw eu gf tf gp mq mp sx ss tc "
                 }
-            }
+            };
         };
+
+        return $.fn.emojioneArea && $.fn.emojioneArea.defaults ? $.fn.emojioneArea.defaults : defaultOptions
```
Comment (mervick) [inline src/var/getDefaultOptions.js]:
```js
return defaultOptions;
```

**Comment by pederjohnsen** [comment]
Must've forgot to re-build after removing debug logs from source... :scream:
Fixed and also made the other requested changes :+1:

**Comment by mervick** [comment]
@pederjohnsen does you tested your new changes?
Now even example not working https://jsfiddle.net/1v03Lqnu/12/
It caused, because `detectVersion()` can't be called earlier than `loadEmojione()`.
First we need to run `loadEmojione()` and only after that use `detectVersion()`.

**Comment by mervick** [comment]
I moved these changes to new branch [dev](https://github.com/mervick/emojionearea/tree/dev) and restored previous version of `master`

**Comment by pederjohnsen** [comment]
Hmm :thinking:, it did work fine for me when I tested, but I only tried with using emojione 3...

I'll take another look tomorrow..

**Comment by pederjohnsen** [comment]
Ah ok I see why now, we manually include the EmojiOne lib before EmojiOneArea in work, and it works then, I guess it fails when it has to include it through EmojiOneArea.. I'll tinker with it a bit further :+1:

**Comment by carpii** [comment]
@mervick, should we submit new pull request to dev branch now, or still master?

This change means my tone_search fixes have been reverted, and only exists in dev branch now

**Comment by mervick** [comment]
@carpii when @pederjohnsen will fix this and create new PR to dev branch I will move it back to master

**Comment by carpii** [comment]
@mervick, ok cool 👍

**Comment by pederjohnsen** [comment]
@mervick should setting `window.emojioneVersion` be required if wanted to use any other emojione version than the default `2.2.7`?

**Comment by pederjohnsen** [comment]
Fix: https://github.com/mervick/emojionearea/pull/185

**Pull request merged**